File: ios/YearlyTracker/WidgetBridgeModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(WidgetBridge, NSObject)
RCT_EXTERN_METHOD(setWidgetPayload:(NSString *)payloadJson)
RCT_EXTERN_METHOD(clearWidgetPayload)
@end

File: ios/YearlyTracker/WidgetBridgeModule.swift
import ExpoModulesCore

public class WidgetBridgeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("WidgetBridge")

    Function("setWidgetPayload") { (payloadJson: String) in
      SharedStore.savePayloadJSON(payloadJson)
    }

    Function("clearWidgetPayload") {
      SharedStore.clearPayload()
    }

    AsyncFunction("setWidgetArtwork") { (sourceUri: String, artworkId: String) in
      SharedStore.writeArtwork(from: sourceUri, artworkId: artworkId)
      return true
    }

    AsyncFunction("clearWidgetArtwork") {
      SharedStore.clearArtwork()
      return true
    }
  }
}
